src/vmspawn/vmspawn.c

```diff
@@ -256,12 +484,46 @@ static int run_virtual_machine(void) {
                 "-m", mem,
                 "-object", "rng-random,filename=/dev/urandom,id=rng0",
                 "-device", "virtio-rng-pci,rng=rng0,id=rng-device0",
-                "-nic", "user,model=virtio-net-pci",
-                "-cpu", "max"
+                "-nic", "user,model=virtio-net-pci"
         );
+        if (!cmdline)
+                return log_oom();
+
+        bool use_vsock = arg_qemu_vsock > 0 && ARCHITECTURE_SUPPORTS_SMBIOS;
+        if (arg_qemu_vsock < 0) {
+                r = qemu_check_vsock_support();
+                if (r < 0)
+                        return log_error_errno(r, "Failed to check for VSock support: %m");
+
+                use_vsock = r;
+        }
+
+        unsigned child_cid = VMADDR_CID_ANY;
+        _cleanup_close_ int child_vsock_fd = -EBADF;
+        if (use_vsock) {
+                if (arg_vsock_cid < UINT_MAX)
+                        child_cid = (unsigned)arg_vsock_cid;
+
+                r = vsock_fix_child_cid(&child_cid, arg_machine, &child_vsock_fd);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to fix CID for the guest vsock socket: %m");
+
+                r = strv_extend(&cmdline, "-device");
+                if (r < 0)
+                        return log_oom();
+
+                log_debug("vhost-vsock-pci,guest-cid=%u,vhostfd=%d", child_cid, child_vsock_fd);
+                r = strv_extendf(&cmdline, "vhost-vsock-pci,guest-cid=%u,vhostfd=%d", child_cid, child_vsock_fd);
+                if (r < 0)
+                        return log_oom();
+        }
+
+        r = strv_extend_strv(&cmdline, STRV_MAKE("-cpu", "max"), false);
+        if (r < 0)
+                return log_oom();
 
         if (arg_qemu_gui) {
-                r = strv_extend_strv(&cmdline, STRV_MAKE("-vga", "virtio"), /* filter_duplicates= */ false);
+                r = strv_extend_strv(&cmdline, STRV_MAKE("-vga", "virtio"), false);
```

Comment:
why this change? we typically want that parameters in function calls that do not tell us anything about what they are are prefix with a comment indicating the parameter name

Reply:
I did this because no other use of strv_extend_strv did this, and I thought it was odd to have it for just one instance. If you would rather I put this for all of them I can do that.

Reply:
I've added it back in but at every callsite for strv_extend_strv in my code.

Reply:
Fixed in 73418f96ff.